### ISSUE TITLE
Bug in get_independencies in BayesianModel

### ISSUE THREAD

**Issue by khalibartan** [pgmpy/pgmpy#607]
I was running BayesianModel on [cancer dataset](www.bnlearn.com/bnrepository/cancer/cancer.bif.gz),
and it gave wrong output. Here is the outputs.

```
In [21]: cancer_model.edges()
Out[21]: 
[('Pollution', 'Cancer'),
 ('Cancer', 'Dyspnoea'),
 ('Cancer', 'Xray'),
 ('Smoker', 'Cancer')]

In [22]: cancer_model.get_independencies()
Out[22]: 
(Dyspnoea _|_ Cancer, Smoker, Xray | Pollution)
(Dyspnoea _|_ Pollution, Cancer, Xray | Smoker)
(Dyspnoea _|_ Pollution, Cancer, Smoker | Xray)
(Pollution _|_ Cancer, Smoker, Xray | Dyspnoea)
(Pollution _|_ Smoker | Cancer)
(Pollution _|_ Dyspnoea, Cancer, Xray | Smoker)
(Pollution _|_ Dyspnoea, Cancer, Smoker | Xray)
(Cancer _|_ Pollution, Smoker, Xray | Dyspnoea)
(Cancer _|_ Dyspnoea, Smoker, Xray | Pollution)
(Cancer _|_ Dyspnoea, Pollution, Xray | Smoker)
(Cancer _|_ Dyspnoea, Pollution, Smoker | Xray)
(Smoker _|_ Pollution, Cancer, Xray | Dyspnoea)
(Smoker _|_ Dyspnoea, Cancer, Xray | Pollution)
(Smoker _|_ Pollution | Cancer)
(Smoker _|_ Dyspnoea, Pollution, Cancer | Xray)
(Xray _|_ Pollution, Cancer, Smoker | Dyspnoea)
(Xray _|_ Dyspnoea, Cancer, Smoker | Pollution)
(Xray _|_ Dyspnoea, Pollution, Cancer | Smoker)

In [23]: cancer_model.is_active_trail('Smoker','Pollution','Cancer')
Out[23]: True
```

From edges it can be seen that 'Smoker', 'Cancer', 'Pollution' is an active trail (given cancer) , so I believe it cannot be independent.

**Comment by khalibartan** [comment]
As per my understanding d-separation implies independence and active-trail implies dependence , correct me if I'm wrong.

**Comment by anaviltripathi** [comment]
I agree with @khalibartan

**Comment by chrisittner** [comment]
Yeah, the method seems to do quite the opposite of what it should :D
Out of six possible conditional independencies in this example, it picked exactly the false ones:

```
>>> a = pgmpy.models.BayesianModel()
>>> a.add_edge('A','B')
>>> a.add_edge('B','C')
>>> a.get_independencies()
(A _|_ B | C)
(C _|_ B | A)
(B _|_ C | A)
(B _|_ A | C)
```

`A` is obviously not independent from `B` no matter what, there is an edge between them. Instead `A_|_C | B` and `C_|_A | B`, which are not in the list.

**Comment by khalibartan** [comment]
Yes actually , code is considering active trail nodes, as independent nodes, it should be other way around.

**Comment by chrisittner** [comment]
Fixed in PR #630
